public/js/controllers/MainController.js

```diff
@@ -1,64 +1,88 @@
- /* jshint esversion: 6 */ 
+ /* jshint esversion: 6 */
+ /* eslint no-param-reassign: ["error", { "props": false }]*/
+
  app.controller('MainController', ['$scope', '$http', '$localStorage',
-   function($scope, $http, $localStorage) {
+   ($scope, $http, $localStorage) => {
+     $scope.noFiles = true;
      $scope.fileReady = false;
-     $scope.afile = {};
-     $scope.files = [];
-
-     // Array that holds the checked files
-     $scope.checked = [];
+     $scope.files = []; // Contains all uploaded file names
+     $scope.checked = []; // Contains all the checked files names
 
-     // Get index fuction
-     // @params file specified
-     // returns the index of the file
-     $scope.fetchIndex = function(file) {
+     /**
+      * createIndex
+      *
+      * sends a http.post to server with the file.
+      * @params {string}
+      * @returns {object} 
+      */
+     $scope.createIndex = (fileName) => {
        $scope.show = true;
-       $scope.fileIndex = $localStorage[file];
-       return file, $scope.fileIndex;
+       const fileContent = JSON.parse($localStorage[fileName]);
+       const data = [fileName, fileContent];
+       $http.post('/createIndex', data).success((index) => {
+         $scope.fileIndex = index;
+       }).error((err) => {
+         $scope.error = 'An error has occurred';
+       });
      };
 
-     // called when a checkbox is checked. 
-     // push or pops a file from the checked array
-     // @params file specified
-     $scope.toogleSelection = function toogleSelection(file) {
-       let indexNo = $scope.checked.indexOf(file);
+     /**
+      * toogleSelection
+      *
+      * called when a checkbox is checked.
+      *
+      * @param {string}
+      */
+     $scope.toogleSelection = (file) => {
+       const indexNo = $scope.checked.indexOf(file);
        if (indexNo > -1) {
          $scope.checked.splice(indexNo, 1);
        } else {
          $scope.checked.push(file);
        }
      };
 
-     // called when user tries to search a file 
-     // returns an array of the results
-     $scope.searchIndex = function() {
+     /**
+      * searchIndex
+      *
+      * passes the selected files and the searched words 
+      * to search function
+      *
+      * @param none
```

Comment:
### Nitpick

Even though it's not explicitly defined in the method signature, it's not entirely accurate that the `search` does not accept any arguments.